### PR TITLE
Gc 20912 zmq fossil

### PR DESCRIPTION
**Pull request by hmak3d**
Precompile binaries so `npm install` no longer needs to recompile it all the time

Currently prebuilt for:
- darwin x64 iojs 1.8.2
- darwin x64 electron 0.25.3 (iojs 1.6.3)
- win32 x64 iojs 1.8.2
- win32 x64 electron 0.25.3 (iojs 1.6.3)

https://grabcad.atlassian.net/browse/GC-20912

### CODE REVIEW

File: prebuilt/install.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+let targ = process.argv[2];
+
+if (! (targ && targ.match(/^(iojs|electron)$/))) {
+    console.error('Bad target: ' + targ);
+    process.exit(1);
+}
+
+let mainDir = path.resolve('build/Release');
+let stashDir = path.resolve(__dirname + '/' + targ + '/' + process.platform + '/' + process.arch + '/Release');
```
Comment (mwhearty) [inline prebuilt/install.js]: a minor nit but should probably use path.join here?

File: prebuilt/install.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+console.log('Installing compilation ' + stashDir + ' -> ' + mainDir);
```
Comment (mwhearty) [inline prebuilt/install.js]: are we guaranteed to always have a console?

File: prebuilt/install.js (hunk 1 above)
Comment (hmak3d) [inline prebuilt/install.js]: Now using `path.join()`

**Comment by hmak3d** [comment]
### New changes

The big change in the latest rebase is to the automation of the precompile step for Electron binaries.  In the previous incarnation, the process was verify clunky ... you had to run `npm` commands on both `eagle-desktop` repo _and_ `zeromq.node` repo just to precompile stuff.  Now everything is self contained in `zeromq.node` repo ... see https://github.com/GrabCAD/zeromq.node/commit/e43ee9553b5de6c882cd91bb181e1b5571660fe0#diff-ce421c73bd85a8c98da620ed0e912235R32

Summary of the overall changes is now:

### Summary of all changes

GC-20912 Precompile zmq.node binary to simplify its usage

* `npm install` from parent module will copy prebuilt Electron binaries [into `zmq/build/Release/`]
* `npm run stash_electron` in zmq module will regenerate the precompiled binaries for use with Electron (i.e., for `eagle-desktop`)
* `npm run stash_iojs` in zmq module will regenerate the precompiled binaries for use with iojs (i.e., for unit tests)

File: prebuilt/install.js (hunk 2 above)
Comment (hmak3d) [inline prebuilt/install.js]: `console` is consistently there when running `npm` scripts

File: prebuilt/install.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+let targ = process.argv[2];
+
+if (! (targ && targ.match(/^(iojs|electron)$/))) {
+    console.error('Bad target: ' + targ);
+    process.exit(1);
+}
+
+let mainDir = path.resolve('build', 'Release');
+let stashDir = path.resolve(__dirname, targ, process.platform, process.arch, 'Release');
```
Comment (Maxodactyl) [inline prebuilt/install.js]: Are we going to switch to path.join as matt suggested?

Comment (hmak3d) [inline prebuilt/install.js]: Yes, the code now uses `path.join()`.  The rebase mashed the new changes into the original commits.  re: https://github.com/GrabCAD/zeromq.node/commit/e43ee9553b5de6c882cd91bb181e1b5571660fe0#diff-d8dc756595bdbe574285bf5f159f35c5R17

**Pull request merged**